pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CastTransformFunction.java

```diff
@@ -93,6 +89,13 @@ public void init(List<TransformFunction> arguments, Map<String, ColumnContext> c
         case "VARCHAR":
           _resultMetadata = sourceSV ? STRING_SV_NO_DICTIONARY_METADATA : STRING_MV_NO_DICTIONARY_METADATA;
           break;
+        case "JSON":
```

Comment:
This is not related to this PR. Just making the order consistent with DataType definition for readability